holoviews/plotting/bokeh/chart.py

```diff
@@ -862,10 +858,11 @@ def _add_color_data(self, ds, ranges, style, cdim, data, mapping, factors, color
 
         # Merge data and mappings
         mapping.update(cmapping)
-        for k, cd in cdata.items():
+        for i, (k, cd) in enumerate(cdata.items()):
             if isinstance(cmapper, CategoricalColorMapper) and cd.dtype.kind in 'uif':
                 cd = categorize_array(cd, cdim)
-            if k not in data or len(data[k]) != next(len(data[key]) for key in data if key != k):
+            # I don't know what this is for but adding the i check makes test pass
```

Comment:
I think you should either give this a bit more thought or ask @philippjfr about it. 

Leaving it as comment in the code makes it hard to find.

Reply:
Adding the i clause fixes this:
```
FAILED test_barplot.py::TestBarPlot::test_box_whisker_multi_level_sorted_alphanumerically - bokeh.util.warnings.BokehUserWarning: ColumnDataSource's columns must be of the same length. Current lengths: ('Category', 9), ('Value', 6), ('width', 6), ('xoffsets', 6)
```
I suspect previously, the tests passed because this is just a BokehUserWarning.

Reply:
I don't understand this line here, maybe @philippjfr or @jlstevens does?

Reply:
Did you figure out why it was needed?

Reply:
Nope